examples/AvroBlogExamples/Program.cs

```diff
@@ -19,31 +19,32 @@
 using System.IO;
 using System.Linq;
 using System.Threading;
-using Avro;
+using System.Threading.Tasks;
 using Avro.Generic;
 using Confluent.Kafka;
-using Confluent.Kafka.Serialization;
+using Confluent.Kafka.AvroSerdes;
+using Confluent.SchemaRegistry;
 
 
 namespace AvroBlogExample
 {
     /// <summary>
     ///     Complete source for the examples programs presented in the blog post:
-    ///     [insert blog URL here]
+    ///     https://www.confluent.io/blog/decoupling-systems-with-apache-kafka-schema-registry-and-avro/
     /// </summary>
     class Program
     {
-        static void ProduceGeneric(string bootstrapServers, string schemaRegistryUrl)
+        async static Task ProduceGeneric(string bootstrapServers, string schemaRegistryUrl)
         {
-            var producerConfig = new ProducerConfig { BootstrapServers = bootstrapServers };
+            using (var schemaRegistry = new CachedSchemaRegistryClient(new SchemaRegistryConfig { SchemaRegistryUrl = schemaRegistryUrl }))

```

Comment:
note that all classes correspond to straightforward concepts, are easy to understand and there aren't any hidden surprises, c.f. 0.11.6 and/or -beta2, for example:
- awaiting `ProduceAsync` could previously block in the avro case.
- it was forbidden to provide the same instance of a serializer to a producer for both key and value.
- the way configuration happened (serializers grabbing the bits they need and passing on), was complex and although 'just worked' was not straightforward.